lib/import/colorectal/providers/st_george/st_george_handler_colorectal.rb

```diff
@@ -0,0 +1,318 @@
+
+            genotype.add_molecular_testing_type_strict(TEST_TYPE_MAP[record.raw_fields['moleculartestingtype']&.downcase&.strip])
+          end
+
+          def assign_test_scope(genotype, record)
+            # extract molecular testing type from the raw record
+            # map molecular testing type and assign to genotype using
+            # add_test_scope method from genocolorectal.rb
+            # return (filled) genotype
+
+            testscope = record.raw_fields['moleculartestingtype']&.downcase&.strip
+            genotype.add_test_scope(TEST_SCOPE_MAP[testscope])
+
+            return genotype if genotype.attribute_map['genetictestscope'].present?
+
+            @logger.error 'ERROR - record with no genetic test scope, ask Fiona for new rules'
+          end
+
+          def fill_genotypes(genotype, record)
+            # process the genes, genotypes and test status for each gene listed in the genotype
+
+            genes_dict = process_genes(genotype, record)
+            handle_test_status(record, genotype, genes_dict)
+          end
+
+          def process_genes(genotype, record)
+            # process the gene names in columns listed and check for matches with the CRC gene regex list
+            # add matched tgenes to the genes_dict
+            # if no matched to crc gene regex list then check for matches in CRC gene map to add to the genes_dict
+            # return genes_dict
+            genes_dict = {}
+
+            columns = if genotype.targeted?
+                        ['gene', 'gene (other)']
+                      elsif genotype.full_screen?
+                        ['gene', 'gene (other)', 'variant_dna', 'test/panel']
+
+                      end
+
+            columns.each do |column|
+              genes = []
+              gene_list = record.raw_fields[column]&.scan(CRC_GENE_REGEX)
+
+              gene_list = process_test_panels(record, gene_list, column) if column == 'test/panel'
+
+              next if gene_list.nil?
+
+              gene_list.each do |gene|
+                CRC_GENE_MAP[gene]&.each do |gene_value|
+                  genes.append(gene_value)
+                end
+              end
+
+              genes_dict[column] = genes.uniq
+            end
+
+            genes_dict
+          end
+
+          def process_test_panels(record, gene_list, column)
+            # extracts panels tested from record
+            # map panels to list of genes in FULL_SCREEN_TESTS_MAP
+            # return list of genes tested in panel
+            panel_genes_list = FULL_SCREEN_TESTS_MAP[record.raw_fields['test/panel']]
+            panel_genes_list&.each do |gene|
+              gene_list.append(gene)
+            end
+
+            r211 = record.raw_fields[column]&.eql?('R211')
+
+            if r211.present?
+              r211_genes = process_r211(record)
+              r211_genes.each do |gene|
+                gene_list.append(gene)
+              end
```

Comment:
You can simply add two arrays rather looping over it and appending here. `gene_list+=r211_genes`

Reply:
Changed in [latest commit](https://github.com/NHSDigital/data_management_system/pull/81/commits/260bc1897440a189bf046c664805725ff4900a5d)